### ISSUE TITLE
Create UserContext

### ISSUE THREAD

**Issue by rcamach7** [lugenx/ecohabit#182]
Integrate an extra context, named UserContext, into the application.

This context should have two states and their respective setState functions:
  - user: Initialize as null
  - token: Set to either null or a token, retrieved from `localStorage.getItem("token")`

Implement a useEffect hook within the UserContext that:
  - Triggers when the token state changes
  - If a token is now present, console.log "fetching user" (actual user fetch functionality will be added later)

Ensure the following are exported:
  - UserContextProvider
  - useUserContext custom hook function

In the App.js file, enclose components with the newly added context. You can reference the other contexts defined as it will closely resemble them.

**Comment by ijklmopffs** [comment]
Hey, I would be willing to work on this

**Comment by rcamach7** [comment]
Assigned @ijklmopffs ! Feel free to reach out for anything

**Comment by rcamach7** [comment]
> Hello, can we connect on discord or telegram?

Yes, you can ping me on our discord server, I am under Mireles#1596

**Comment by lugenx** [comment]
hi @ijklmopffs , can we get a status update on this?

**Comment by lugenx** [comment]
i will remove the assignment from this issue since we have yet to get a response from the potential contributor. If anyone would like to get assigned, feel free to comment below.

**Comment by victorchrollo14** [comment]
hey @lugenx , I'll add this up.

**Comment by lugenx** [comment]
Awesome! Assigned. Thank you! 👍
